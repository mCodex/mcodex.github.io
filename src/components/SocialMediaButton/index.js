/* @flow */

import React from 'react';
import { Icon } from 'semantic-ui-react';

type Props = {
  onClick?: Function,
  iconName: string,
  uri: string
};

export default function SocialMediaButton({ onClick, iconName, uri }: Props) {
  return (
    <a
      id={`${iconName}Button`}
      href={uri}
      target="_blank"
      rel="noopener noreferrer"
      onClick={() => onClick()}
    >
      <Icon name={iconName} size="big" style={{ color: 'white' }} />
    </a>
  );
}

SocialMediaButton.defaultProps = {
  onClick: () => ({})
};
