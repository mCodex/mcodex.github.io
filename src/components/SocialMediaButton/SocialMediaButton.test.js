import React from 'react';
import renderer from 'react-test-renderer';
import SocialMediaButton from '.';

describe('Home', () => {
  it('Snapshot Testing', () => {
    const wrapper = renderer.create(<SocialMediaButton />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('Testing if screen is rendering', () => {
    const wrapper = renderer.create(<SocialMediaButton />).toJSON();
    expect(wrapper).toBeTruthy();
  });

  it('Testing if button is clickable', () => {
    const spy = sinon.spy();

    const wrapper = mount(<SocialMediaButton onClick={spy} iconName="linkedin" />);

    const linkedinButton = wrapper.find('#linkedinButton').first();

    linkedinButton.simulate('click');
    expect(spy.calledOnce).toBe(true);
  });
});
