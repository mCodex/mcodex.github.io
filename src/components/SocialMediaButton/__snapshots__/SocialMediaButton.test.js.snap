// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot Testing 1`] = `
<a
  id="undefinedButton"
  onClick={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  <i
    aria-hidden="true"
    className="big icon"
    onClick={[Function]}
    style={
      Object {
        "color": "white",
      }
    }
  />
</a>
`;
